syft/pkg/cataloger/cpp/parse_conanlock.go

```diff
@@ -0,0 +1,54 @@
+package cpp
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+// integrity check
+var _ common.ParserFn = parseConanlock
+
+type conanLock struct {
+	GraphLock struct {
+		Nodes map[string]struct {
+			Ref string
+		}
+	} `json:"graph_lock"`
+}
+
+// parseConanlock is a parser function for conan.lock contents, returning all packages discovered.
+func parseConanlock(_ string, reader io.Reader) ([]*pkg.Package, []artifact.Relationship, error) {
+	pkgs := []*pkg.Package{}
+	var cl conanLock
+	if err := json.NewDecoder(reader).Decode(&cl); err != nil {
+		return nil, nil, err
+	}
+	for _, node := range cl.GraphLock.Nodes {
+		if len(node.Ref) > 0 {
+			// ref: pkga/0.1@user/testing
+			splits := strings.Split(strings.Split(node.Ref, "@")[0], "/")
+			if len(splits) < 2 {
+				continue
+			}
+			pkgName, pkgVersion := splits[0], splits[1]
+			pkgs = append(pkgs, &pkg.Package{
+				Name:         pkgName,
+				Version:      pkgVersion,
+				Language:     pkg.CPP,
+				Type:         pkg.ConanPkg,
+				MetadataType: pkg.ConanaMetadataType,
+				Metadata: pkg.ConanMetadata{
```

Comment:
capturing the name and version isn't really necessary. There is one or two other catalogers that do this for implementation-specific limitations, however, in this case it's redundant.

That being said, there is other information in a `conan.lock` file that could be useful to capture (such as `options`).  It's probably worth taking a look at what other information could be provided and adding that to the metadata.

If we don't want to add any more elements, my recommendation now would be to not have a metadata struct at all.

Reply:
@wagoodman I can add `options` `path` and `context` as metadata fields and remove `name` and `version` here

Then I'll update the `conanfile` to include `nil` for the metadata

Reply:
@wagoodman it also looks like `name` and `version` are tied to the method for generating a PURL for a given `conan` package. It might be useful to keep those fields so we can agnostically generate the PURL for either a `lockfile` or `conanfile` even if the information is redundant for the package struct:

https://github.com/anchore/syft/blob/main/syft/pkg/conan_metadata.go

Is there another space you would want this method?

Reply:
ahh, we're reusing the same metadata for logically a different things. I don't think we should do this. That is, the `Metadata` is reserved for things that are specific from what this package was parsed from (I didn't see the reuse on the first review pass). We should probably be introducing a new metadata type to capture this information.